src/bin/parser-test.rs

```diff
@@ -21,8 +19,8 @@ pub struct TestResults {
     tests_failed: Vec<(usize, usize, String)>,
 }
 
-fn main() -> Result<()> {
-    let mut results = TestResults {
+fn main() {
+    let mut results = GlobalTestResults {
```

Comment:
I don't know that "Global" tells us anything here that "TestResults" didn't already.  But "global" has the disadvantage of connotations of a singleton or global variable.

Reply:
`GlobalTestResults` are actually an aggregation of test results. It is used to keep track on how many of the tests fails or succeeded. But i agree that the naming is a bit wrong here..  I think a TotalResults would be better.